### PR TITLE
CLDC-673: Organisation Page frontend

### PR DESCRIPTION
**Pull request by baarkerlounger**
- [x] Tab Navigation component
- [x] Organisation details page
- [x] Organisation users page
- [x] Users last sign in trackable
- [x] Add organisation link to header bar
- [x] Feature test for tab interaction
- [x] User table styling
- [x] Link for Invite User button -> Doesn't currently work as we don't currently have a create new user page (will be added as part of https://digital.dclg.gov.uk/jira/browse/CLDC-307)
- [x] Helpers for user page

image
image

### CODE REVIEW

File: app/javascript/styles/_tab-navigation.scss [hunk 1]
```diff
@@ -0,0 +1,76 @@
+.app-tab-navigation {
+  @include govuk-font(19, $weight: bold);
+  @include govuk-responsive-margin(6, "bottom");
+}
+
+.app-tab-navigation__list {
+  @include govuk-clearfix;
+  // left: govuk-spacing(-3);
```
Comment (baarkerlounger) [inline app/javascript/styles/_tab-navigation.scss]: @paulrobertlloyd not sure what this is supposed to do exactly but seems to error on negative values. Any ideas?

**Comment by kosiakkatrina** [comment]
Is this a govuk component like that? 
image

The `details` and `users` don't seem very intuitive. I'd expect one of them to be highlighted etc. if it's currently open

Also. the highlight on the nav bar selected page disappears if you navigate to change password, or details/users within the organisation tab
image

File: app/javascript/styles/_tab-navigation.scss (hunk 1 above)
Comment (paulrobertlloyd) [reply app/javascript/styles/_tab-navigation.scss]: What’s erroring, Sass? Are we using the latest version of `govuk-frontend` – `3.14.0` added support for negative values.

File: app/components/tab_navigation_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<nav class="app-tab-navigation" aria-label="sub menu">
+  <ul class="app-tab-navigation__list">
+    <% items.each do |item| %>
+      <li class="app-tab-navigation__item">
+        <% if item.fetch(:current, false) || current_page?(strip_query(item.fetch(:url))) %>
+          <%= govuk_link_to item[:name], item[:url], class: 'app-tab-navigation__link', aria: { current: 'page' } %>
+        <% else %>
+          <%= govuk_link_to item[:name], item[:url], class: 'app-tab-navigation__link' %>
+        <% end %>
+      </li>
```
Comment (paulrobertlloyd) [inline app/components/tab_navigation_component.html.erb]: The component has slightly different markup to the one on Apply I sent you:

```suggestion
      <% if item.fetch(:current, false) || current_page?(strip_query(item.fetch(:url))) %>
        <li class="app-tab-navigation__item app-tab-navigation__item--current">
          <%= govuk_link_to item[:name], item[:url], class: 'app-tab-navigation__link', aria: { current: 'page' } %>
        </li>
      <% else %>
        <li class="app-tab-navigation__item">
          <%= govuk_link_to item[:name], item[:url], class: 'app-tab-navigation__link' %>
        </li>
      <% end %>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
Could we call this component `SubNavigationComponent`/`app-sub-navigation`? ‘Tab navigation is too easily confused with the design system’s ‘Tabs’ component.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -40,6 +40,7 @@
           component.navigation_item(text: 'Case logs', href: '/case_logs')
         elsif
           component.navigation_item(text: 'Your account', href: '/users/account')
+          component.navigation_item(text: 'Organisation', href: "/organisations/#{current_user.organisation.id}")
```
Comment (paulrobertlloyd) [inline app/views/layouts/application.html.erb]:
```suggestion
          component.navigation_item(text: 'Your organisation', href: "/organisations/#{current_user.organisation.id}")
          component.navigation_item(text: 'Your account', href: '/users/account')
```

File: app/views/layouts/organisations.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+<% content_for :content do %>
+  <h1 class="govuk-heading-l">
+    Your Organisation
+  </h1>
+
+  <%= render TabNavigationComponent.new(items: [
+    { name: t('Details'), url: details_organisation_path(@organisation) },
+    { name: t('Users'), url: users_organisation_path(@organisation) },
+  ]) %>
+
```
Comment (paulrobertlloyd) [inline app/views/layouts/organisations.html.erb:17]: Can we add the following visually hidden title here. I’ve been meaning to add this to the prototype.

```suggestion

  <h2 class="govuk-visually-hidden"><%= PAGE_TITLE %></h2>
```

Where `PAGE_TITLE` is either ‘Details’ or ‘Users’

File: app/javascript/styles/_tab-navigation.scss (hunk 1 above)
Comment (baarkerlounger) [reply app/javascript/styles/_tab-navigation.scss]: Ah no, we're on 3.13.0, will see if I can bump

**Comment by paulrobertlloyd** [comment]
> Is this a govuk component like that?

@kosiakkatrina It is not, but it is a derivative of one used on many services. The MoJ Design System has [something similar](https://design-patterns.service.justice.gov.uk/components/sub-navigation/). Unfortunately the design system lacks components for navigation (outside of the header, which itself isn’t great), and GDS have left it up to individual teams to come up with their own solutions. Hopefully one day the design system will include a set of navigation patterns to save duplication and varied implementations.

**Comment by baarkerlounger** [comment]
> Could we call this component SubNavigationComponent/app-sub-navigation? ‘Tab navigation is too easily confused with the design system’s ‘Tabs’ component.

I've based it on what DfE have called theirs. I think that's better for consistency, particularly if this does ever get pushed upstream?

**Comment by paulrobertlloyd** [comment]
> I think that's better for consistency

That ship sailed a long time ago! Not a massive issue, will update the prototype to match.

File: app/views/layouts/organisations.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<% content_for :before_content do %>
+  <%= govuk_back_link(
+    text: 'Back',
+    href: :back,
+  ) %>
+<% end %>
+
+<% content_for :content do %>
+  <h1 class="govuk-heading-l">
+    Your Organisation
```
Comment (paulrobertlloyd) [inline app/views/layouts/organisations.html.erb]:
```suggestion
    Your organisation
```

File: app/views/layouts/organisations.html.erb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  </h1>
+
+  <h2 class="govuk-visually-hidden"><%= content_for(:tab_title) %></h2>
+
+  <%= render TabNavigationComponent.new(items: [
+    { name: t('Details'), url: details_organisation_path(@organisation) },
+    { name: t('Users'), url: users_organisation_path(@organisation) },
+  ]) %>
```
Comment (paulrobertlloyd) [inline app/views/layouts/organisations.html.erb]: Sub heading should be below the navigation:

```suggestion
  <%= render TabNavigationComponent.new(items: [
    { name: t('Details'), url: details_organisation_path(@organisation) },
    { name: t('Users'), url: users_organisation_path(@organisation) },
  ]) %>

  <h2 class="govuk-visually-hidden"><%= content_for(:tab_title) %></h2>
```

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]

**Review by paulrobertlloyd — APPROVED** [review]
Thanks @baarkerlounger!

**Pull request merged**